selfservice/strategy/oidc/provider_generic_oidc.go

```diff
@@ -103,10 +108,43 @@ func (g *ProviderGenericOIDC) Claims(ctx context.Context, exchange *oauth2.Token
 		return nil, errors.WithStack(ErrIDTokenMissing)
 	}
 
+	return g.ClaimsFromIdToken(ctx, raw)
+}
+
+func (g *ProviderGenericOIDC) ClaimsFromIdToken(ctx context.Context, rawIdToken string) (*Claims, error) {
+	p, err := g.provider(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return g.verifyAndDecodeClaimsWithProvider(ctx, p, rawIdToken)
+}
+
+func (g *ProviderGenericOIDC) ClaimsFromAccessToken(ctx context.Context, accessToken string) (*Claims, error) {
```

Comment:
Using an access token to authenticate a user is extremely dangerous. The problem is that an evil user eve can create a legitimate OAuth2 App, trick good user alice to use that OAuth2 app, and then use the access token eve received from that app at Ory Kratos, even though the alice never authenticated in Ory Kratos.

Therefore, we need to axe this flow. It can only ever be possible to use `ClaimsFromIDToken()`

Reply:
Yes, we have discussed this earlier.
Will remove `access_token` option.

Reply:
Nice, thanks :)

Reply:
done